src/senselab/audio/tasks/features_extraction/praat_parselmouth.py

```diff
@@ -1,143 +1,1377 @@
+                tmin = parselmouth.praat.call(vuv_table, "Get value", i + 1, "tmin")
+                tmax = parselmouth.praat.call(vuv_table, "Get value", i + 1, "tmax")
+                snd_segment = snd.extract_part(float(tmin), float(tmax))
+
+                PowerCepstrogram = parselmouth.praat.call(snd_segment, "To PowerCepstrogram", 60, 0.002, 5000, 50)
+                # PowerCepstrogram (60-Hz pitch floor, 2-ms time step, 5-kHz maximum frequency,
+                # and pre-emphasis from 50 Hz)
+
+                try:
+                    CPP_Value = parselmouth.praat.call(
+                        PowerCepstrogram,
+                        "Get CPPS...",
+                        "no",
+                        0.01,
+                        0.001,
+                        60,
+                        330,
+                        0.05,
+                        "parabolic",
+                        0.001,
+                        0,
+                        "Straight",
+                        "Robust",
+                    )
+                    # Subtract tilt before smoothing = “no”; time averaging window = 0.01 s;
+                    # quefrency averaging window = 0.001 s;
+                    # Peak search pitch range = 60–330 Hz; tolerance = 0.05; interpolation = “Parabolic”;
+                    # tilt line frequency range = 0.001–0 s (no upper bound);
+                    # Line type = “Straight”; fit method = “Robust.”
+                except Exception as e:
+                    current_frame = inspect.currentframe()
+                    if current_frame is not None:
+                        current_function_name = current_frame.f_code.co_name
+                        logger.error(f'Error in "{current_function_name}": \n' + str(e))
+                    CPP_Value = np.nan
+
+                if not np.isnan(CPP_Value) and CPP_Value > 4:
+                    cpp_list.append(CPP_Value)
+
+        # Calculate Final Features
+        if cpp_list:
+            CPP_array = np.array(cpp_list)
+            CPP_mean = np.mean(CPP_array)
+            CPP_std = np.std(CPP_array)
+        else:
+            CPP_mean = np.nan
+            CPP_std = np.nan
+
+        # Return Result
+        return {"mean_cpp": CPP_mean, "std_dev_cpp": CPP_std}
+
+    except Exception as e:
+        current_frame = inspect.currentframe()
+        if current_frame is not None:
+            current_function_name = current_frame.f_code.co_name
+            logger.error(f'Error in "{current_function_name}": \n' + str(e))
+        return {"mean_cpp": float("nan"), "std_dev_cpp": float("nan")}
+
+
+def measure_formants(
+    snd: Union[parselmouth.Sound, Path, Audio], floor: float, ceiling: float, frame_shift: float
+) -> Dict[str, float]:
+    """Extract Formant Frequency Features.
+
+    Function to extract formant frequency features from a given sound object. This function is adapted from default
+    Praat code to work with Parselmouth.
```

Comment:
may want to add the description of pulses here to indicate that the extraction happens per pitch period.